### DISCUSSION TITLE
Ask for exporting an api of validating when clause

### DISCUSSION THREAD

**Discussion by XiLaiTL** [microsoft/vscode-discussions#994] [Extension Development QnA] [answered]
I want to validate the [when clause](https://code.visualstudio.com/api/references/when-clause-contexts) dynamically in my extension, when I use it as a config for user. 
However, the function [validateWhenClauses](https://github.com/microsoft/vscode/blob/main/src/vs/platform/contextkey/common/contextkey.ts#L640) which can be used to validate the `when clause` was not exported in @type/vscode.
How can I do this?

**Comment by isidorn** [comment, marked as answer]
This is a fair request. But I do not think we have it on the plan right now.
fyi @ulugbekna

**Reply by XiLaiTL** [reply]
I wanna to make the api of my extension more similar to the keysmapping api like this:
```json
{
    "command":"","when":"","args":{}
}
```

In fact, this idea of exporting when-validation api will provide support to many custom extensions
